presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -173,25 +173,37 @@ public synchronized void freeSpill(long bytes)
         spillSpaceTracker.free(bytes);
     }
 
-    public synchronized void setMemoryPool(MemoryPool pool)
+    public void setMemoryPool(MemoryPool pool)
     {
-        requireNonNull(pool, "pool is null");
-        if (memoryPool == pool) {
-            // Don't unblock our tasks and thrash the pools, if this is a no-op
-            return;
+        // this method acquires the monitors of both queryMemoryContext and
+        // this instance, and there can be a deadlock if they are not
+        // acquired in the same order.
+
+        // the thread calling this method should first acquire the lock
+        // on queryMemoryContext as that's the order driver threads
+        // acquire the monitors of these objects when they reserve memory
+        // through the child memory context -> ... -> root memory context -> query context
```

Comment:
This comments are sentences. Capitalize and add `.` where necessary

Reply:
Additionally, explain that we need the lock for `this` here, and some of the methods we call here will attempt to acquire a lock on `queryMemoryContext` (explain how).